src/expression.rs

```diff
@@ -3,6 +3,11 @@ use crate::Data;
 use serde_json::Value;
 use std::collections::HashSet;
 
+#[derive(Debug)]
+pub struct Ambiguous;
```

Comment:
Do we need this type? You can just return `PartialApplyResult::Ambiguous`

Reply:
yes we need it, else `PartialResult` would have to become
```type PartialResult = Result<Value, PartialApplyOutcome>```

and since `PartialApplyOutcome` has two 2 enums, would need to unnecessarily handle extra case or use Err(_), and I didn't want to use _ for this case matching